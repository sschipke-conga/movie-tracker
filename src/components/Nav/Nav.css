header {
  display:flex;
  justify-content: space-between;
  align-items:center;
  height:80px;
  position: fixed;
  top: 0;
  width: 100%;
  z-index:2;
  border-bottom:1px solid gray
}

.logo {
  height:45px
}

.header__logo {
  font-size: 35px;
  min-width: 250px;
  letter-spacing: 1px;
  display:flex;

}

.red {
  color:red;
  letter-spacing:-3px;
}

.header__nav {
  width:80%;
  display:flex;
}

.NavLink{
  text-decoration: none;
  width:250px;
  font-size:20px;
  line-height: 45px;
}

.selectedLink {
  text-transform: uppercase;
}

.NavLink:hover {
  animation-name: btn-micro;
  animation-duration:0.5s;
}

.user-avatar{
  height:25px;
  margin-right:9px;
  background-color: rgba(255,255,255,0);
}

.header__user-menu{
  margin-top:10px;
  width:250px;
  margin-right:15px
}

.inner {
  display:flex;
  justify-content: flex-end;
 
  background: red;
  color:white;
  padding: 1em;
  border-radius: 10px;
  width: 250px;
  clip-path: circle(10% at 85% 50%);
  transition: all .5s ease-in-out;
  cursor: pointer;
}

.inner:hover {
  clip-path: circle(75%);
  background: black;
}

.inner h1 {
  color: white;
  margin: 0;
}

.inner p {
  color: white;
  font-size: .8rem;
}


.inner:hover span {
  color: rgba(255,255,255,0)
}

.header__user-name{
  background-color:transparent;
  margin-right:30px;

}


@keyframes btn-micro {
  0% { transform: rotate(0deg)}
  25%{ transform: rotate(5deg) }
  50%{ transform: rotate(0deg) }
  75%{ transform: rotate(-5deg) }
  100%{transform: rotate(0deg)}

}