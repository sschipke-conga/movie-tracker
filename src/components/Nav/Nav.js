import React from 'react';
import { NavLink, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import './Nav.css';
import PropTypes from 'prop-types';
import avatar from '../../images/user-avatar.svg'
const Nav = ({user}) => {

  return (
    <header>
      <h1 className='header__logo'>Movie<span className='red'>Snag</span></h1>
      <div className='header__nav'>
        <NavLink exact to='/' className='NavLink' activeClassName="selectedLink" >  
          Home
        </NavLink>
        <NavLink exact to='/now_playing' className='NavLink' activeClassName="selectedLink">  
          Now Playing
        </NavLink>
        <NavLink exact to='/upcoming' className='NavLink' activeClassName="selectedLink">  
          Coming Soon
        </NavLink>
        <NavLink exact to='/favorites' className='NavLink' activeClassName="selectedLink">  
          Favorites
        </NavLink>
      </div>
      <div className="header__user-menu">
        <div className="inner">
          <h3 className='header__user-name'>
            {user.name ? (<>{user.name} <a href='/login'>Logout</a></>): (<a href='/login'>Login</a>)}
          </h3>
          <img className='user-avatar' alt='line drawing of a torso'src={avatar} />
          <p></p>
        </div>
      </div>
    </header>
  )
}

const mapStateToProps = state => ({
  user: state.user
})



export default connect(mapStateToProps)(Nav);
