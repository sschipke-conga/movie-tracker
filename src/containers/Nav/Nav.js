import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './Nav.css';
import PropTypes from 'prop-types';
import avatar from '../../images/user-avatar.svg';
import logo from '../../images/MovieSnagLogo.png';

export const Nav = ({ user }) => {
  return (
    <header>
      <div className="header__nav">
        <Link to="./" className="">
          <h1 className="header__logo">
            <img alt="MovieSnag logo" src={logo} className="logo"/>
            ovie
            <span className="red">
              Snag
            </span>
          </h1>
        </Link>
        <NavLink exact to="/" className="NavLink" activeClassName="selectedLink" >  
          Home
        </NavLink>
        <NavLink exact to="/now_playing" className="NavLink" activeClassName="selectedLink">
          Now Playing
        </NavLink>
        <NavLink exact to="/upcoming" className="NavLink" activeClassName="selectedLink">
          Coming Soon
        </NavLink>
        <NavLink exact to="/favorites" className="NavLink" activeClassName="selectedLink">
          Favorites
        </NavLink>
      </div>
      <div className="header__user-menu">
        <div className="inner">
          <h3 className="header__user-name">
            {user.name ? (<>{user.name} <a href="/login">Logout</a></>) : (<a href="/login">Login</a>)}
          </h3>
          <img className="user-avatar" alt="line drawing of a torso" src={avatar} />
          <p></p>
        </div>
      </div>
    </header>
  );
};

export const mapStateToProps = (state) => ({
  user: state.user,
});

export default connect(mapStateToProps)(Nav);

Nav.propTypes = {
  user: PropTypes.object.isRequired
}
