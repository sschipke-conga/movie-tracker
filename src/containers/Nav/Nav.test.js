import React from 'react';
import { Nav, mapStateToProps} from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<Nav user={{}} />);
    expect(wrapper).toMatchSnapshot();
  })
})

describe('mapStateToProps', () => {
  it('should return an object with a user in it', () => {
    const mockUser = {
      id: 5,
      name: 'Greg',
      email: 'g@dev.com',
      password: 'password1234'
    }
    const mockState = {user: mockUser}

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(mockState)
  })
})

describe('Nav', () => {
  it('should render a logout button if a user is passed in', () => {
    const mockUser = {
      id: 5,
      name: 'Greg',
      email: 'g@dev.com',
      password: 'password1234'
    }
    const wrapper = shallow(<Nav user={mockUser}/>)
    expect(wrapper).toMatchSnapshot()
  })
})