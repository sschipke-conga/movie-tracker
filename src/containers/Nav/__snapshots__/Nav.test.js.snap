// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<header>
  <div
    className="header__nav"
  >
    <Link
      className=""
      to="./"
    >
      <h1
        className="header__logo"
      >
        <img
          alt="MovieSnag logo"
          className="logo"
          src="MovieSnagLogo.png"
        />
        ovie
        <span
          className="red"
        >
          Snag
        </span>
      </h1>
    </Link>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/now_playing"
    >
      Now Playing
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/upcoming"
    >
      Coming Soon
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/favorites"
    >
      Favorites
    </NavLink>
  </div>
  <div
    className="header__user-menu"
  >
    <div
      className="inner"
    >
      <h3
        className="header__user-name"
      >
        <a
          href="/login"
        >
          Login
        </a>
      </h3>
      <img
        alt="line drawing of a torso"
        className="user-avatar"
        src="user-avatar.svg"
      />
      <p />
    </div>
  </div>
</header>
`;

exports[`Nav should render a logout button if a user is passed in 1`] = `
<header>
  <div
    className="header__nav"
  >
    <Link
      className=""
      to="./"
    >
      <h1
        className="header__logo"
      >
        <img
          alt="MovieSnag logo"
          className="logo"
          src="MovieSnagLogo.png"
        />
        ovie
        <span
          className="red"
        >
          Snag
        </span>
      </h1>
    </Link>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/now_playing"
    >
      Now Playing
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/upcoming"
    >
      Coming Soon
    </NavLink>
    <NavLink
      activeClassName="selectedLink"
      className="NavLink"
      exact={true}
      to="/favorites"
    >
      Favorites
    </NavLink>
  </div>
  <div
    className="header__user-menu"
  >
    <div
      className="inner"
    >
      <h3
        className="header__user-name"
      >
        Greg
         
        <a
          href="/login"
        >
          Logout
        </a>
      </h3>
      <img
        alt="line drawing of a torso"
        className="user-avatar"
        src="user-avatar.svg"
      />
      <p />
    </div>
  </div>
</header>
`;
